engine/font/src/dmsdk/font/text_layout.h

```diff
@@ -0,0 +1,212 @@
+    float       m_Width;        // width of the glyph bounding box
+    float       m_Height;       // height of the glyph bounding box
+
+    // TODO: See if we can remove these
+    float  m_Advance;     // LEGACY SHAPING ONLY!
+    float  m_LeftBearing; // LEGACY SHAPING ONLY!
+
+    uint32_t    m_Codepoint;    // Not always available if there was a substitution
+    uint16_t    m_GlyphIndex;   // index into the font
+    uint16_t    m_Cluster;      // index into the original text (i.e. into codepoints)
+};
+
+/*#
+ * Represents a line of glyphs
+ * @struct
+ * @name TextLine
+ * @member m_Width [type: float] Width of the line
+ * @member m_Index [type: uint16_t] Index into the list of glyphs
+ * @member m_Length [type: uint16_t] Number of glyphs to render
+ */
+struct TextLine
+{
+    float    m_Width;
+    uint16_t m_Index;
+    uint16_t m_Length;
+};
+
+/*#
+ * Represents a line of glyphs
+ * @struct
+ * @name TextLayoutSettings
+ * @member m_Size [type: float] The desired size of the font
+ * @member m_Width [type: float] Max layout width. used only when line_break is non-zero
+ * @member m_Leading [type: float]
+ * @member m_Tracking [type: float]
+
+ * @member m_Padding [type: uint32_t] Legacy: Padding for monospace, glyphbank fonts
+ * @member m_LineBreak [type: uint8_t:1] Allow line breaks
+ * @member m_Monospace [type: uint8_t:1] Legacy: Is the font a monospace font. Current: should be set on the font in the font collection!
+ */
+struct TextLayoutSettings
+{
+    float     m_Size;
+    float     m_Width;
+    float     m_Leading;
+    float     m_Tracking;
+
+    uint32_t    m_Padding;
+    uint8_t     m_LineBreak:1;
+    uint8_t     m_Monospace:1;
+};
+
+
+/*#
+ * Create a text layout using a font collection
+ * if successful, the caller must call TextLayoutFree() on the layout
+ * @name TextLayoutCreate
+ * @param collection [type: HFontCollection] the font collection
+ * @param codepoints [type: uint32_t*] an array of codepoints
+ * @param num_codepoints [type: uint32_t] number of codepoints in the array
+ * @param settings [type: TextLayoutSettings*] the settings used for rendering
+ * @param layout [type: HTextLayout*] the output text layout
+ * @return result [type: TextResult] the result. TEXT_RESULT_OK if successful
+ */
+TextResult TextLayoutCreate(HFontCollection collection,
+                            uint32_t* codepoints, uint32_t num_codepoints,
+                            TextLayoutSettings* settings, HTextLayout* layout);
+
+/*#
+ * Frees a previously created layout
+ * @name TextLayoutFree
+ * @param layout [type: HTextLayout] the text layout
+ */
+void TextLayoutFree(HTextLayout layout);
+
+/*#
+ * Get the glyph count in the layout
+ * @name TextLayoutGetGlyphCount
+ * @param layout [type: HTextLayout] the text layout
+ * @return count [type: uint32_t] the number of glyphs in the layout
+ */
+uint32_t TextLayoutGetGlyphCount(HTextLayout layout);
```

Comment:
Is it worth to merge `TextLayoutGetGlyphCount` and `TextLayoutGetGlyphs` and return both array and array size?